### PR TITLE
Provide preference mapping for Java Execution Environments -> Runtimes

### PR DESCRIPTION
**Pull request by snjeza**
Requires https://github.com/eclipse/eclipse.jdt.ls/pull/1309

Signed-off-by: Snjezana Peco <snjezana.peco@redhat.com>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -432,6 +432,39 @@
...
+              },
+              "javadoc": {
+                "type": "string",
+                "label": "JDK javadoc."
+              }
+            }
+          },
+          "default": [],
+          "description": "Java Execution Environments -> Runtimes.",
+          "scope": "window"
```
Comment (fbricon) [inline package.json]: should be machine scoped

File: package.json [hunk 2]
```diff
@@ -432,6 +432,39 @@
...
+            "type": "object",
+            "default": {},
+            "required": [
+              "path",
+              "name"
+            ],
+            "properties": {
+              "name": {
+                "type": "string",
+                "label": "Java Execution Environment name. Must be unique"
```
Comment (fbricon) [inline package.json]: this should be an enumerated list of all allowed values

File: package.json (hunk 1 above)
Comment (snjeza) [reply package.json]: Done.

File: package.json (hunk 2 above)
Comment (snjeza) [reply package.json]: Fixed. See https://github.com/redhat-developer/vscode-java/pull/1207/commits/f8b6a181340d4e77c2fb6804e92668d29ecf55fe#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R464

Comment (fbricon) [reply package.json]: That's a welcome change nut not what I meant. Names should be limited to JavaSE-1.8 ... JavaSE-13, all values of execution environments supported in eclipse

Comment (snjeza) [reply package.json]: Do you  want the following:
```
"java.configuration.runtimes": {
          "type": "array",
          "label": "Java Execution Environments.",
          "items": {
            "type": "object",
            "default": {},
            "required": [
              "path",
              "name"
            ],
            "properties": {
              "name": {
                "type": "string",
                "enum": [
                    "JavaSE-8",
                    "JavaSE-9",
                    "JavaSE-10",
                    "JavaSE-10+",
                    "JavaSE-11",
                    "JavaSE-12",
                    "JavaSE-13"
                  ],
                "label": "Java Execution Environment name. Must be unique"
              },
              "path": {
                "type": "string",
                "label": "JDK path.\nOn Windows, backslashes must be escaped, i.e.\n\"path\":\"C:\\\\Program Files\\\\Java\\\\jdk1.8.0_161\""
              },
              "sources": {
                "type": "string",
                "label": "JDK sources."
              },
              "javadoc": {
                "type": "string",
                "label": "JDK javadoc."
              },
              "default": {
                "type": "boolean",
                "label": "Is default runtime? Only one runtime can be default."
              }
            },
            "additionalProperties": false
          },
          "default": [],
          "description": "Java Execution Environments -> Runtimes.",
          "scope": "machine"
        }
```

Comment (fbricon) [reply package.json]: All that is here, no more no less:
Screen Shot 2019-12-20 at 1 39 56 AM

**Comment by snjeza** [comment]
@fbricon I have updated the PR.

File: package.json (hunk 2 above)
Comment (snjeza) [reply package.json]: Done.

**Comment by fbricon** [comment]
Final syntax looks like :

"java.configuration.runtimes": [
        {
            "name": "JavaSE-1.8",
            "path": "/usr/local/jdk1.8.0_201"
        },
        {
            "name": "JavaSE-11",
            "path": "/usr/local/jdk-11.0.3",
            "sources" : "/usr/local/jdk-11.0.3/lib/src.zip",
            "javadoc" : "https://docs.oracle.com/en/java/javase/11/docs/api",
            "default":  true
        },
        {
            "name": "JavaSE-12",
            "path": "/usr/local/jdk-12.0.2"
        },
        {
            "name": "JavaSE-13",
            "path": "/usr/local/jdk-13"
        }
    ]
Runtime name must be one of:
"J2SE-1.5",
"JavaSE-1.6",
"JavaSE-1.7",
"JavaSE-1.8",
"JavaSE-9",
"JavaSE-10",
"JavaSE-11",
"JavaSE-12",
"JavaSE-13"

List will be updated with each supported release of the JDK

**Pull request merged**